lib/libzpool/uzfs_mgmt.c

```diff
@@ -305,6 +325,7 @@ uzfs_open_dataset_init(const char *ds_name, zvol_state_t **z)
 	if (error)
 		goto disown_free;
 
+	/* Meta-object may not exist if dataset wasn't created by uzfs */
 	error = dmu_object_info(os, ZVOL_META_OBJ, &doi);
 
 	error |= zap_lookup(os, ZVOL_ZAP_OBJ, "metavolblocksize", 8, 1,
```

Comment:
we should start throwing error

Reply:
it depends. We have three options:

 * We could either return an error to disallow opening zvol which was not created by uzfs
 * We could leave the code in this func as it is and let zrepl deal with missing meta obj (i.e. it could create it or ignore the zvol)
 * We could create meta obj here if we find that it is missing (though I'm not 100% sure if it is possible to modify zvol in this context - when opening it)

I can adjust the PR according to your preference.

Reply:
As this code should come only in uzfs, we should throw error in my opinion if ZVOL_META_OBJ is not there.

Reply:
I will do that 👍